src/UIKit/UIGestureRecognizer.cs

```diff
@@ -41,9 +40,9 @@ void OnDispose ()
 		{
 			var copyOfRecognizers = recognizers;
 			var savedHandle = Handle;
-			recognizers = null;
+			recognizers.Clear ();
 			
-			if (copyOfRecognizers == null)
+			if (copyOfRecognizers is null)
```

Comment:
This is an interesting case I didn't think of before: `copyOfRecognizers` will never be null (because `recognizers` never will be), which means that the below code will always be executed. However, because the `copyOfRecognizers` and the `recognizers` variable point to the same dictionary instance, the code below won't actually do anything because it will be operating on a cleared dictionary.

The purpose of this code is to keep the existing dictionary around, clearing the field (nulling it out in the old code), and then iterate over all the entries in the existing dictionary to remove them.

There are a couple of ways to keep this behavior, one that comes to mind is:

1. Assign a new dictionary to the `recognizers` variable (instead of clearing it):

    ```cs
    recognizers = new Dictionary<Token, IntPtr> ();
    ```

2. Check if `copyOfRecognizers` is empty instead of `null`:

    ```cs
    if (copyOfRecognizers.Count == 0)
        return;
    ```